src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/IncrementingEventCounter.cs

```diff
@@ -56,6 +57,9 @@ public void Increment(double increment = 1)
 
         public override string ToString() => $"IncrementingEventCounter '{Name}' Increment {_increment}";
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "The DynamicDependency will preserve the properties of IncrementingCounterPayload")]
+        [DynamicDependency(DynamicallyAccessedMemberTypes.PublicProperties, typeof(IncrementingCounterPayload))]
```

Comment:
Same question about IncrementingEventCounterPayloadType.

Reply:
I used your recommendation to annotate Write<T> and then add the DynamicDependency as needed for FX eventsource types